src/harness/reference_models/ppa_model/utils.py

```diff
@@ -0,0 +1,154 @@
+#    Copyright 2016 SAS Project Authors. All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+import numpy as np
+from area import area
+import geojson
+from shapely.geometry import shape
+from shapely.ops import cascaded_union
+from src.harness.util import getRandomLatLongInPolygon
+from collections import defaultdict
+from datetime import datetime
+import random
+import json
+import os
+import glob
+
+
+def to_dist_bear_vincenty(**kwargs):
+  """Placeholder to Call Vincenty, will be removed when the Vincenty is moved to reference models"""
+  pass
+
+
+def great_circle(**kwargs):
+  latitude = kwargs.pop('latitude')
+  longitude = kwargs.pop('longitude')
+  distance = kwargs.pop('distance') # In Km.
+  azimuth = kwargs.pop('azimuth')
+  calc = np.vectorize(to_dist_bear_vincenty)
+  lat, lon, reverse = calc(latitude, longitude, distance, azimuth)
+  return {"latitude": lat, "longitude": lon}
+
+
+def check_hole(ppa):
+  if hasattr(ppa, 'interiors'):
+    if ppa.interiors:
+      interior_coords = [interior.coords[:] for interior in ppa.interiors]
+      polygon = geojson.Polygon(interior_coords)
+      area_hole = area(polygon)
+      if area_hole < 500000.0:
+        return cascaded_union([ppa, shape(polygon)])
+  return ppa
+
+
+def convert_shape(polygon):
+  ppa = shape(polygon).buffer(0)
+  return ppa
+
+
+def smooth(x, window_len=15):
+  s = np.r_[x[-1:-window_len / 2:-1], x, x[window_len / 2:0:-1]]
+  w = np.hamming(window_len)
```

Comment:
Thanks for noting that! How about something like `s = np.r_[x[-window_len:],x,x[:window_len]]`?